pypeit/waveimage.py

```diff
@@ -50,7 +50,7 @@ def __init__(self, tslits_dict, tilts, wv_calib, spectrograph, maskslits,
 
         # MasterFrame
         masterframe.MasterFrame.__init__(self, self.frametype, master_key,
-                                         master_dir=master_dir, reuse_masters=reuse_masters)
+                                         master_dir, reuse_masters=reuse_masters)
```

Comment:
What if someone is running out of a notebook and does not want to write out a masterframe? I think master_key and master_dir should probably remain optional arguments, since otherwise you need to pass them in with None to just run but not write.

Reply:
I had the same thought.  While these are now requite to init `masterframe`, that is 
a meta class (i.e. it cannot be instantiated on its own, so won't).

The base classes, e.g. `waveimage` can be instantiated without providing
`master_dir`, i.e. they pass in `None` be default.

My thinking is that you can't build the default masterframe name for a file
without both `master_key` and `master_dir` so might as well make them
both required inputs (allowing for `None`).

Reply:
I guess I don't see why having them be optional for the child, but required for the parent is such a problem. I guess the question is what will the save_master method if you pass in None. I think it will crash, rather than choose some default filename. 

I think it is easy to figure out that you don't have to pass in something then figure out that you need to instantiate to None.